File: src/code/javascript/graph/dfs_recursive.js
const fn = (graph) => {
    const dfs = (node) => {
        let ans = 0
        // TODO: logic
        for (const neighbor of graph[node]) {
            if (!seen.has(neighbor)) {
                seen.add(neighbor)
                ans += dfs(neighbor)
            }
        }

        return ans
    }

    let seen = new Set([START_NODE])

    return dfs(START_NODE)
}

File: src/code/ruby/graph/dfs_recursive.rb
require 'set'

def fn(graph)
  seen = Set.new([START_NODE])

  dfs = lambda do |node|
    ans = 0
    # TODO: logic
    graph[node].each do |neighbor|
      unless seen.include?(neighbor)
        seen.add(neighbor)
        ans += dfs.call(neighbor)
      end
    end
    ans
  end

  dfs.call(START_NODE)
end
